src/pages/Watch.tsx

```diff
@@ -176,25 +183,25 @@ export default function WatchPage() {
     setEditLyrics(!editLyrics);
   }
 
-  async function handleChangePitch(e, data) {
-    return changePitch(data.value, videoMode);
+  async function handleChangePitch(_: unknown, data: any) {
+    return changePitch(Number(data.value), videoMode!);
   }
 
-  async function handleSynced(mode) {
+  async function handleSynced(mode: library.CaptionsMode) {
     init(mode);
   }
 
-  async function init(customCaptionsMode) {
+  async function init(customCaptionsMode?: library.CaptionsMode) {
     try {
       window.scrollTo({ top: 0, behavior: "smooth" });
-      setVideoURL(null);
-      setLang(null);
-      setCaptionsURL(null);
+      setVideoURL(undefined);
```

Comment:
why do you prefer undefined over null?

Reply:
TypeScript seems to prefer undefined, and makes it easier to use with shortcuts when defining object types like `{ foo?: string }`, whereas with null it would have to be `{ foo: string | null }`.

The other really good argument I've heard for using undefined is that it's unavoidable that you'll have to deal it as a value in your code, since undefined is the default returned for an uninitialized variable, or when an object property or array index doesn't exist, and using undefined instead of null means you only have to worry about one special "null" value instead of two.